target/linux/ramips/dts/mt7620a_asus_rt-ac54u.dts

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "asus,rt-ac54u", "ralink,mt7620a-soc";
+	model = "Asus RT-AC54U";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "rt-ac54u:blue:power";
+			gpios = <&gpio0 9 GPIO_ACTIVE_LOW>;
+		};
+
+		usb {
+			label = "rt-ac54u:blue:usb";
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&ohci_port1>, <&ehci_port1>;
+			linux,default-trigger = "usbport";
+		};
+
+		wifi {
+			label = "rt-ac54u:blue:wifi";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0{
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Can this be increased?

Reply:
How do I know how much to increase?

Reply:
Have a look at the datasheets (if you find some) and/or try increasing and see whether it's stable. You may also check which other recently added devices have the same flash, as we started to request this quite frequently.
Most of the older devices just always copied 10 MHz from some other DTS, and this is considerably slower than necessary.

Reply:
datasheet says it's 50MHz when doing normal. I will start with that

Reply:
You may try that with and without `m25p,fast-read;`. Just benchmark with a simple dd if the device is booting. Normally, only certain steps are possible for the frequency, so you might only see a performance change if you double etc.

You may also try 80 MHz with fast-read if the specs state it's supported and recovery is not too painful.

Reply:
specs say read 50MHz
fast read 104MHz 
[datasheet](https://www.mxic.com.tw/Lists/Datasheet/Attachments/7333/MX25L12835E,%203V,%20128Mb,%20v1.3.pdf)  (Table 13. AC CHARACTERISTICS)

Reply:
You may try 100 MHz for fun, but I don't think it will boot.
Based on you evaluation, I'd take the 50 with fast-read ...